File: javascript/difference-of-squares/difference-of-squares.js
//Difference of Squares exercise.

export class Squares {
  constructor(square) {
    this.square = square
  }
  
  range(){
    let arr = []
    let i = 1
    while(i <= this.square){
      arr.push(i)
      i++
    }
    return arr
  }
  
  sum(){
    return this.range().reduce((a,b) => a+b)
  }

  get sumOfSquares() {
     return this.range()
      .map(i => Math.pow(i, 2))
      .reduce((a,b) => a+b)
  }

  get squareOfSum() {
   return Math.pow(this.sum(), 2)
  }

  get difference() {
    return this.squareOfSum - this.sumOfSquares
  }
}

File: ruby/difference-of-squares/difference_of_squares.rb
# frozen_string_literal:  true

# Diference of squares exercism
class Squares
  attr_reader :number

  def initialize(number)
    @number = number
  end

  def sum_of_squares
    (1..number).map { |key| key.pow(2) }.sum
  end

  def square_of_sum
    (1..number).sum.pow(2)
  end

  def difference
    square_of_sum - sum_of_squares
  end
end
